shared/js/content-scripts/click-to-load.js

```diff
@@ -462,6 +462,58 @@
             return frame
         }
 
+        /*
+         * Tweaks an embedded YouTube video element ready for reinsertion to the
+         * page.
+         *
+         * @param {Element} videoElement
+         * @returns {Function?} onError
+         *   Function to be called if the video fails to load.
+         */
+        async adjustYouTubeVideoElement (videoElement) {
+            let onError = null
+
+            if (!videoElement.src) {
+                return onError
+            }
+            const url = new URL(videoElement.src)
+            const { hostname: originalHostname } = url
+
+            // Upgrade video to YouTube's "privacy enhanced" mode, but fall back
+            // to standard mode if the video fails to load.
+            // Note:
+            //  1. Changing the iframe's host like this won't cause a CSP
+            //     violation on Chrome, see https://crbug.com/1271196.
+            //  2. The onError event doesn't fire for blocked iframes on Chrome.
+            if (originalHostname !== 'www.youtube-nocookie.com') {
+                url.hostname = 'www.youtube-nocookie.com'
+                onError = (event) => {
+                    url.hostname = originalHostname
+                    videoElement.src = url.href
+                    event.stopImmediatePropagation()
+                }
+            }
+
+            // Ensure the video doesn't auto-play.
+            const allowString = videoElement.getAttribute('allow') || ''
+            const allowed = new Set(allowString.split(';').map(s => s.trim()))
+            allowed.delete('autoplay')
```

Comment:
Are you sure we want this? In other browsers we explicitly fix autoplay and this has had a explicit user interaction.

Reply:
Yea, I'm sure since we discussed it in a PFR. With the current implementation the placeholder gives no clue about the video, so autoplay on click isn't a nice user experience. If we add the preview image + title in the future, we'll then do the  reverse and explicitly enable auto-play.